server/admin/admin_server.go

```diff
@@ -72,12 +74,28 @@ func (s *Server) getTreeImpl(ctx context.Context, request *trillian.GetTreeReque
 func (s *Server) CreateTree(ctx context.Context, request *trillian.CreateTreeRequest) (*trillian.Tree, error) {
 	tree, err := s.createTreeImpl(ctx, request)
 	if err != nil {
-		return nil, errors.WrapError(err)
+		return nil, svrerrors.WrapError(err)
 	}
 	return tree, err
 }
 
 func (s *Server) createTreeImpl(ctx context.Context, request *trillian.CreateTreeRequest) (*trillian.Tree, error) {
+	// Test that the tree's private key exists. If it doesn't, generate it.
+	if _, err := s.registry.SignerFactory.NewSigner(ctx, request.GetTree()); err != nil {
+		if errors.ErrorCode(err) != errors.NotFound {
```

Comment:
A way we could do this is to add key generation settings to the CreateTreeRequest. Actually, that might be even better than behaving differently depending on the supplied private key.